### ISSUE TITLE
model.train_on_batch(accuracy = True) bug with Theano backend

### ISSUE THREAD

**Issue by jmhessel** [fchollet/keras#1404]
Hi Folks,

I switched some code over from the Tensorflow backend to the Theano backend, and I got a strange formatting error. The code is as follows:

```
trainL, acc = model.train_on_batch(cXtrain, cYtrain, accuracy = True)
print "loss={:.3f}, acc={:.3f}".format(aveLoss, acc)
```

when using the TensorFlow backend, acc is a np.float32. When using the theano backend, acc is a numpy ndarray. I don't have time to dig through to figure out what's happening but I figured I'd report this.

Jack
